### DISCUSSION TITLE
Model rotates around a random point in Filament viewer

### DISCUSSION THREAD

**Discussion by udayk07** [google/filament#4983] [Q&A]
While loading the given test model [test_bb.zip](https://github.com/google/filament/files/7733312/test_bb.zip) in [ filament gltf viewer](https://google.github.io/filament/viewer/) it renders fine. When trying to rotate/pan model, the model rotates around a random pivot point ( hope it is bound box issue ).  

But, in other gltf viewers [Viewer 1](https://github.khronos.org/glTF-Sample-Viewer-Release/) , [Viewer 2](https://gltf-viewer.donmccurdy.com/) the model rotates as expected. 

I am trying to achieve the same behaviour in Android ( with sceneform/filament ) . How to rotate model as it is without rotating around a random point ? Is there any options available in filament to resolve ?

**Comment by romainguy** [comment]
The bounding box doesn't affect rotation. There could be an issue with the transforms defined the gltf (or how we interpret them). Also note that gltf_viewer uses an orbital camera, the camera rotates around the object but the object itself doesn't rotate.

**Reply by udayk07** [reply]
@romainguy Thanks for reply. 
Came across the pr https://github.com/google/filament/pull/4973 today. It seems,  this is same as that. Will that help to resolve the rotation behaviour ?

**Reply by pixelflinger** [reply]
No it's unrelated.

**Comment by udayk07** [comment]
@romainguy 

We are trying to implement marker based detection in AR ( with sceneform - Android ). While loading the model on marker detected, users can not rotate around its center instead it rotates around some orgin ( as u mentioned, it may be transform issue ). Is there any workaround available to resolve this issue ?

https://user-images.githubusercontent.com/78372917/148204264-82eea9ba-eabf-433e-a3f0-a70f8f99c6f9.mp4

**Comment by pixelflinger** [comment]
What you're seeing is just a problem in our web viewer. Try with the desktop gltf_viewer and it works fine.

I don't think it's a random point, it's just that the pivot point is in 0 and the object is not.

**Reply by udayk07** [reply]
@pixelflinger 
We are implementing it in marker based ar with filament ( sceneform ) in Android platform. While doing so, we are facing above behaviour. 

`it's just that the pivot point is in 0 and the object is not` Is there any workaround to re-center object even though object is not in orgin so that object rotation works as expected ?
